dive_core/shader_disassembly.cpp

```diff
@@ -30,16 +37,52 @@ bool Disassemble(const uint8_t*                             shader_memory,
     return false;
 }
 
-void Disassembly::Init(const uint8_t* data, uint64_t address, size_t max_size, ILog* log_ptr) {}
-
-std::string Disassembly::GetInstructionText(uint32_t index) const
+void Disassembly::Init(const uint8_t* data, uint64_t address, size_t max_size, ILog* log_ptr)
 {
-    return "";
+    std::string         disasm;
+    struct shader_stats stats;
+    {
+        enum debug_t debug = PRINT_RAW;
+        disasm_a3xx_set_debug(debug);
+
+        // TODO(bjoeris): this should come from the file instead of hard-coded.
+        uint32_t chip = 0x43050b00;
```

Comment:
i added g_sGPU_variant in pm4_info.cpp (generated from generatePm4Info_adreno.py) we could probably add chip_id there as well